### PR TITLE
Improve wallet disconnect transaction modal state

### LINKED ISSUE #911: Make button "connect to wallet" for any button that initiates a transaction.
If no wallet is connected, any button that initiates a transaction doesn't do anything when clicked. This is a confusing UX.

### proposal

Make every button that initiates a transaction open the "connect wallet" modal if no wallet is connected.

### possible implementation

Might be able to build a single TransactorButton component that handles this logic internally.

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Closes #911 

To me, having the OK button be enabled and the okText as 'Connect wallet to ...' and then opening the wallet select would be ideal if not for the fact that this causes a page reload and therefore closes the modal on the user unexpectedly once they connect their wallet. (reload occurs because setting wallet writes to local storage). 

So then, I think the next best option is just disabling the button with the 'Connect wallet to ...' text, and make the user back out to connect their wallet. 100% open to feedback or other ideas here though.

For every transaction modal that users can get to without a wallet connected, I've implemented this logic. 

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/173198766-10551d5a-309f-40a0-9c97-9868c50e81de.mp4

## Acceptance checklist

- [ ] I have evaluated the
      [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines)
      for this PR.
- [ ] I have tested this PR in
      [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/shared/TransactionModal.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import { Modal, ModalProps } from 'antd'
+import { NetworkContext } from 'contexts/networkContext'
 
 import { ThemeContext } from 'contexts/themeContext'
 import { PropsWithChildren, useContext } from 'react'
 
 type TransactionModalProps = PropsWithChildren<
   ModalProps & {
     transactionPending?: boolean
+    connectWalletText?: string
```
Comment (aeolianeth) [inline src/components/shared/TransactionModal.tsx:11]: **praise** love the prop idea!

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@johnnyd-eth this probably works for me.

But a reload need not occur - there's a solution to prevent this. I agree it's out of scope of this PR.

There's a few other buttons that we should port this functionality to, which was my original intention with this issue:
- reconfigure modal
- edit project details drawer
- transfer unclaimed tokens
- add to balance

So that we don't duplicate logic, I suggested we have a `TransactorButton` component that we use anywhere we create a transaction. let me know what you think

**Comment by johnnyd-eth** [comment]
> @johnnyd-eth this probably works for me.
> 
> But a reload need not occur - there's a solution to prevent this. I agree it's out of scope of this PR.
> 
> There's a few other buttons that we should port this functionality to, which was my original intention with this issue:
> 
> * reconfigure modal
> * edit project details drawer
> * transfer unclaimed tokens
> * add to balance
> 
> So that we don't duplicate logic, I suggested we have a `TransactorButton` component that we use anywhere we create a transaction. let me know what you think

@aeolianeth Yeah good call, I did miss the 'transfer' and 'add to balance' buttons - I created a `TransactorButton` component for them. I think the cleanest solution is still just what I've done with `TransactorModal` since `antd` modals can't take buttons as props.

As for the other instances you've mentioned, they can't be seen or accessed with the wallet disconnected so I didn't think they were worth doing anything for.

**Comment by johnnyd-eth** [comment]
Screen Shot 2022-06-12 at 9 34 28 am

Also snuck in a fix to this layout

**Comment by aeolianeth** [comment]
@johnnyd-eth ah yeah understood.

Merging! Thanks for working on this, really nice UX imrpovement.

**Review by aeolianeth — APPROVED** [review]

**Pull request merged**